src/uct/ib/dc/dc_mlx5.c

```diff
@@ -1149,6 +1172,21 @@ ucs_status_t uct_dc_mlx5_iface_fc_grant(uct_pending_req_t *self)
     return status;
 }
 
+void uct_dc_mlx5_iface_fc_grant_send(uct_dc_fc_request_t *dc_req)
+{
+    ucs_status_t status;
+
+    status = uct_dc_mlx5_iface_fc_grant(&dc_req->super.super);
+    if (status == UCS_ERR_NO_RESOURCE){
+        uct_dc_mlx5_ep_do_pending_fc(ucs_derived_of(dc_req->super.ep,
+                                                    uct_dc_mlx5_ep_t),
+                                     dc_req);
+    } else {
+        ucs_assertv_always(status == UCS_OK, "Failed to send FC grant msg: %s",
+                           ucs_status_string(status));
```

Comment:
assert_always will terminate process without any chance for error handling. is it ok?
@brminich what do you think? may be just report error and allow error handling to make it's work?

Reply:
we could just remove the assertion, but we don't have a chance to invoke error callabkac, since we don't know the endpoint on which we should invoke it

Reply:
i think it is ok to assert on local failures, as we can consider it as a software error